Fix dnsmasq v5 to v6 config migration

MichaIng opened the pull request:
## Thank you for your contribution to the Pi-hole Community!

Please read the comments below to help us consider your Pull Request.

We are all volunteers and completing the process outlined will help us review your commits quicker.

**Please make sure you**

 1. Base your code and PRs against the repositories developmental branch.
 2. [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits as we enforce the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
 3. [Sign](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all your commits as they must have verified signatures
 4. File a pull request for any change that requires changes to [our documentation](https://docs.pi-hole.net/) at our [documentation repo](https://github.com/pi-hole/docs) 

---

**What does this PR aim to accomplish?:**

The dnsmasq config files were removed in `remove_old_dnsmasq_ftl_configs()`, before they were tried to be migrated via `migrate_dnsmasq_configs()`, and hence most settings were lost during v5 to v6 update.

**How does this PR accomplish the above?:**

This commit renames and adjussts `remove_old_dnsmasq_ftl_configs()` to move dnsmasq config files into the migration directory instead, to be picked up by `migrate_dnsmasq_configs()` later.

Before the commit, no migration is done. Not sure why there is not a single log line about it, but different topic:
```
  [✓] Installing latest logrotate script
  [i] man not installed
  [✓] New password set
  [i] Testing if systemd-resolved is enabled
```
After the commit, making `/opt/pihole/update.sh` calling the local installer from this PR, the migration is done, including the web UI password, the upstream DNS etc.
```
  [✓] Installing latest logrotate script
  [i] man not installed
  [✓] Migrating Pi-hole configuration to version 6
    Migrating config to Pi-hole v6.0 format
    Reading legacy config files from /etc/pihole/pihole-FTL.conf
       PIDFILE: Empty path is not possible, using default
       SETUPVARSFILE: Empty path is not possible, using default
       MACVENDORDB: Empty path is not possible, using default
       GRAVITYDB: Empty path is not possible, using default
       WEBROOT: Empty path is not possible, using default
       WEBHOME: Empty path is not possible, using default
       API_INFO_LOG: Empty path is not possible, using default
       WEBDOMAIN: Empty path is not possible, using default
    Moving /etc/pihole/pihole-FTL.conf to /etc/pihole/migration_backup_v6/pihole-FTL.conf
    Migrating config from /etc/pihole/setupVars.conf
    setupVars.conf:WEBPASSWORD -> Setting webserver.api.pwhash to 54a6673c814b32309806ff1067ede0155de1af6c0550d6ce51d6f62ba76e4101
    setupVars.conf:BLOCKING_ENABLED -> Setting dns.blocking.active to true
    setupVars.conf:TEMPERATURE_LIMIT -> Not set
    setupVars.conf:TEMPERATURE_UNIT -> Not set
    setupVars.conf:WEBUIBOXEDLAYOUT -> Not set
    setupVars.conf:WEBTHEME -> Not set
    setupVars.conf:PIHOLE_DNS_1 -> Setting dns.upstreams[1] = 1.1.1.1
    setupVars.conf:PIHOLE_DNS_2 -> Setting dns.upstreams[2] = 1.0.0.1
    setupVars.conf:PIHOLE_DOMAIN -> Not set
    setupVars.conf:DNS_FQDN_REQUIRED -> Setting dns.domainNeeded to true
    setupVars.conf:DNS_FQDN_REQUIRED -> Setting dns.expandHosts to true
    setupVars.conf:DNS_bogusPriv -> Not set
    setupVars.conf:DNSSEC -> Setting dns.dnssec to false
    setupVars.conf:PIHOLE_INTERFACE -> Setting dns.interface to eth0
    setupVars.conf:HOSTRECORD -> Not set
    setupVars.conf:DNSMASQ_LISTENING -> Setting dns.listeningMode to LOCAL
    setupVars.conf:REV_SERVER_CIDR -> Not set
    setupVars.conf:REV_SERVER_TARGET -> Not set
    setupVars.conf:REV_SERVER_DOMAIN -> Not set
    setupVars.conf:REV_SERVER_* -> Not set (found invalid/incomplete parameters)
    setupVars.conf:DHCP_ACTIVE -> Not set
    setupVars.conf:DHCP_START -> Not set
    setupVars.conf:DHCP_END -> Not set
    setupVars.conf:DHCP_ROUTER -> Not set
    setupVars.conf:DHCP_LEASETIME -> Not set
    setupVars.conf:DHCP_IPv6 -> Not set
    setupVars.conf:DHCP_RAPID_COMMIT -> Not set
    setupVars.conf:queryLogging -> Not set
    setupVars.conf:GRAVITY_TMPDIR -> Not set
    setupVars.conf:WEB_PORTS -> Not set
    Moved /etc/pihole/setupVars.conf to /etc/pihole/migration_backup_v6/setupVars.conf
    setupVars.conf migration complete
    Moving /etc/pihole/custom.list to /etc/pihole/migration_backup_v6/custom.list
    Config initialized with webserver ports 8080 (HTTP) and 443 (HTTPS), IPv6 support is enabled
    Wrote config file:
     - 152 total entries
     - 143 entries are default
     - 9 entries are modified
     - 0 entries are forced through environment
    Config file written to /etc/pihole/pihole.toml
  [i] Testing if systemd-resolved is enabled
```

**Link documentation PRs if any are needed to support this PR:**

<!--- Replace this with a link to your documentation PR  -->

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

PromoFaux commented:
Thanks @MichaIng - presumably you've tested this on a v5 snapshot of some kind?

I'll review this in a sec - I just need to test another theory on the [tweak/migrate](https://github.com/pi-hole/pi-hole/tree/tweak/migrate) branch

MichaIng commented:
Yeah, I did test it on a latest Pi-hole v5. But I am also investigating another issue, that `pihole.toml` somehow exists already while it should not.

yubiuser commented:
Please rebase once https://github.com/pi-hole/pi-hole/pull/5969 has been merged.

PromoFaux commented:
@MichaIng rebase

MichaIng commented:
Let me run a final test/comparison ...

MichaIng commented:
So set up a fresh Pi-hole v5 instance to test/compared: I downloaded `v5.18.4` tag installer, added a `git reset --hard <latest_v5_tag>` to the end of the `make_repo` function (tag based on whether it's core or web), to be on master branch but latest v5 state, and changed the FTL URL to `https://github.com/pi-hole/FTL/releases/download/v5.25.2` (latest v5 tag).

This installed Pi-hole v5 as intended, and compared to my previous instances, without an additional Git learning curve when cloning a tag instead of a branch in the first place 😄:
```console
root@VM-Bookworm:~# pihole -v
  Pi-hole version is v5.18.4 (Latest: v6.0.3)
  web version is v5.21 (Latest: v6.0.1)
  FTL version is v5.25.2 (Latest: v6.0.2)
root@VM-Bookworm:~# l /etc/dnsmasq.d/
total 8.0K
-rw-r--r-- 1 root root 1.4K Feb 22 22:00 01-pihole.conf
-rw-r--r-- 1 root root 2.2K Feb 22 22:00 06-rfc6761.conf
root@VM-Bookworm:~# l /etc/pihole/
total 14M
-rw-r--r-- 1 root   root      0 Feb 22 22:00 custom.list
-rw-r--r-- 1 pihole pihole    0 Feb 22 22:00 dhcp.leases
-rw-r--r-- 1 root   root    651 Feb 22 22:00 dns-servers.conf
-rw-rw-r-- 1 pihole pihole 7.4M Feb 22 22:00 gravity.db
-rw-rw-r-- 1 pihole pihole  92K Feb 22 22:00 gravity_old.db
-rw-r--r-- 1 root   root    849 Feb 22 22:00 install.log
-rw-r--r-- 1 root   root   2.6M Feb 22 22:00 list.1.raw.githubusercontent.com.domains
-rw-r--r-- 1 root   root     95 Feb 22 22:00 list.1.raw.githubusercontent.com.domains.sha1
-rw-r--r-- 1 root   root     65 Feb 22 22:00 local.list
-rw-r--r-- 1 root   root    289 Feb 22 22:00 logrotate
-rw-r--r-- 1 pihole pihole 3.2M Feb 22 22:00 macvendor.db
drwxr-xr-x 2 root   root   4.0K Feb 22 22:00 migration_backup
-rw-rw-r-- 1 pihole root    127 Feb 22 22:00 pihole-FTL.conf
-rw-rw-r-- 1 pihole pihole  80K Feb 22 22:01 pihole-FTL.db
-rw-r--r-- 1 root   root    339 Feb 22 22:00 setupVars.conf
-rw-r--r-- 1 root   root    324 Feb 22 22:00 versions
root@VM-Bookworm:~#
```
Now upgrading via `pihole -up`, but changing `/opt/pihole/update.sh` to call the latest `development` branch installer, instead of the local `master` branch one:
```sh
sed -i '\|basic-install.sh --reconfigure|c\curl -sSfO https://raw.githubusercontent.com/pi-hole/pi-hole/refs/heads/development/automated%20install/basic-install.sh; bash basic-install.sh --reconfigure --unattended || \\' /opt/pihole/update.sh
pihole -up
```
I was offered to disable Lighttpd, btw. Looks good, but the last output line is confusing:
```console
...
  [✓] Migrating Pi-hole configuration to version 6
    Migrating config to Pi-hole v6.0 format
    Reading legacy config files from /etc/pihole/pihole-FTL.conf
       PIDFILE: Empty path is not possible, using default
       SETUPVARSFILE: Empty path is not possible, using default
       MACVENDORDB: Empty path is not possible, using default
       GRAVITYDB: Empty path is not possible, using default
       WEBROOT: Empty path is not possible, using default
       WEBHOME: Empty path is not possible, using default
       API_INFO_LOG: Empty path is not possible, using default
       WEBDOMAIN: Empty path is not possible, using default
    Moving /etc/pihole/pihole-FTL.conf to /etc/pihole/migration_backup_v6/pihole-FTL.conf
    Migrating config from /etc/pihole/setupVars.conf
    setupVars.conf:WEBPASSWORD -> Setting webserver.api.pwhash to ccfc60b185a182d78c955df21654bdbb4b99bb444931af4faecaa4ea8a4968e8
...
  [i] The install log is located at: /etc/pihole/install.log
  [✓] Update complete!

Core version is v6.0.3 (Latest: v6.0.3)
Web version is v6.0.1 (Latest: v6.0.1)
FTL version is v6.0.2 (Latest: v6.0.2)
root@VM-Bookworm:~# l /etc/pihole/migration_backup_v6/
total 8.0K
-rw-rw---- 1 pihole pihole   0 Feb 22 22:00 custom.list
-rw-rw---- 1 pihole pihole 127 Feb 22 22:00 pihole-FTL.conf
-rw-rw---- 1 pihole pihole 339 Feb 22 22:00 setupVars.conf
```
So the migration works, but as can be seen, `01-pihole.conf` and `06-rfc6761.conf` have not been migrated, as they were removed before the migration is done.

Now doing the same from v5, but using the installer from this PR:
```sh
sed -i '\|basic-install.sh --reconfigure|c\curl -sSfO https://raw.githubusercontent.com/MichaIng/pi-hole/refs/heads/fix/config-migration/automated%20install/basic-install.sh; bash basic-install.sh --reconfigure --unattended || \\' /opt/pihole/update.sh
pihole -up
```
Same result/output, but the dnsmasq configs have been migrated as intended:
```console
...
  [✓] Migrating Pi-hole configuration to version 6
    Migrating config to Pi-hole v6.0 format
    Reading legacy config files from /etc/pihole/pihole-FTL.conf
       PIDFILE: Empty path is not possible, using default
       SETUPVARSFILE: Empty path is not possible, using default
       MACVENDORDB: Empty path is not possible, using default
       GRAVITYDB: Empty path is not possible, using default
       WEBROOT: Empty path is not possible, using default
       WEBHOME: Empty path is not possible, using default
       API_INFO_LOG: Empty path is not possible, using default
       WEBDOMAIN: Empty path is not possible, using default
    Moving /etc/pihole/pihole-FTL.conf to /etc/pihole/migration_backup_v6/pihole-FTL.conf
    Migrating config from /etc/pihole/setupVars.conf
    setupVars.conf:WEBPASSWORD -> Setting webserver.api.pwhash to ccfc60b185a182d78c955df21654bdbb4b99bb444931af4faecaa4ea8a4968e8
...
  [✓] Update complete!

Core version is v6.0.3 (Latest: v6.0.3)
Web version is v6.0.1 (Latest: v6.0.1)
FTL version is v6.0.2 (Latest: v6.0.2)
root@VM-Bookworm:~# l /etc/pihole/migration_backup_v6/
total 16K
-rw-rw---- 1 pihole pihole 1.4K Feb 22 22:00 01-pihole.conf
-rw-rw---- 1 pihole pihole 2.2K Feb 22 22:00 06-rfc6761.conf
-rw-rw---- 1 pihole pihole    0 Feb 22 22:00 custom.list
-rw-rw---- 1 pihole pihole  127 Feb 22 22:00 pihole-FTL.conf
-rw-rw---- 1 pihole pihole  339 Feb 22 22:00 setupVars.conf
```
